### PR TITLE
Refine CDK Custom Exec Policy - Linking Envs

### PR DESCRIPTION
**Pull request by noah-paige**
### Feature or Bugfix

- Bugfix

### Detail
- Add S3 Permissions so CDK Execution Policy is able to access CDK Asset Staging Bucket and auto create Pivot Role if enabled
- Add details to instructions of linking environment to make Prerequisite steps more clear
Screenshot 2023-08-09 at 12 25 27 PM

### Relates
- #625 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: deploy/cdk_exec_policy/cdkExecPolicy.yaml
Comment (dlpzx) [inline deploy/cdk_exec_policy/cdkExecPolicy.yaml:48]: <3

File: deploy/cdk_exec_policy/cdkExecPolicy.yaml
Comment (dlpzx) [inline deploy/cdk_exec_policy/cdkExecPolicy.yaml:232]: Wow, how did we miss this? Thanks!

**Review by dlpzx — APPROVED** [review]

**Comment by dlpzx** [comment]
Looks good. I think we can merge to `main` for modularization I would wait until the frontend PRs are merged

**Comment by nikpodsh** [comment]
@dlpzx, all frontend modularization PRs are targeting to the other branch (`modularization/frontend`). The are only a couple of PRs to `modularization-main`. I will create a PR to collect all bugfixes from the main

**Pull request merged**
